app/code/community/Phoenix/CashOnDelivery/Block/Totals/Abstract.php

```diff
@@ -44,7 +44,7 @@ public function initTotals()
     {
         $this->_prepareTotals();
 
-        if ($this->_totalObject && $this->_totalObject->getCodFee()) {
+        if ($this->_totalObject && $this->_totalObject->getCodFee() && Mage::getStoreConfigFlag('payment/phoenix_cashondelivery/display_zero_fee')) {
```

Comment:
This will not work... If codFee check returns false the last && will not be validated. If the codFee Check returns true and display_zero_fee is set to false the total block is not added even if a fee is set.